src/main/scala/org/ergoplatform/Input.scala

```diff
@@ -41,6 +43,8 @@ class UnsignedInput(val boxId: BoxId, val extension: ContextExtension) {
     case _ => false
   }
 
+  override def hashCode(): Int = boxId.hashCode
```

Comment:
Same problem here, need to use deepHashCode

Reply:
Fixed.